controllers/servicediscovery/servicediscovery_controller.go

```diff
@@ -477,7 +472,35 @@ func (r *Reconciler) configureDNSConfigMap(ctx context.Context, cr *submarinerv1
 		return goerrors.New("the lighthouse DNS Service ClusterIP is not set")
 	}
 
-	return r.updateLighthouseConfigInConfigMap(ctx, cr, configMapNamespace, configMapName, lighthouseDNSService.Spec.ClusterIP)
+	err = r.updateLighthouseConfigInConfigMap(ctx, cr, DefaultCoreDNSNamespace, CoreDNSName, lighthouseDNSService.Spec.ClusterIP)
+
+	if apierrors.IsNotFound(err) {
+		// Some providers may not use the exact "coredns" name but use it as a suffix, eg RKE "rke2-coredns".
+		configMaps := &corev1.ConfigMapList{}
+
+		listErr := r.GeneralClient.List(ctx, configMaps, controllerClient.InNamespace(DefaultCoreDNSNamespace))
+		if listErr != nil {
+			return errors.Wrapf(err, "error listing ConfigMaps in %q", DefaultCoreDNSNamespace)
+		}
+
+		suffix := "-" + CoreDNSName
+
+		for i := range configMaps.Items {
+			cm := &configMaps.Items[i]
+
+			_, hasCorefile := cm.Data[Corefile]
+			if strings.HasSuffix(cm.Name, suffix) && hasCorefile {
```

Comment:
Based on [this comment](https://github.com/submariner-io/lighthouse/issues/1602#issuecomment-2518254501), we could we just check for the existence of the `Corefile` data entry. @vthapar @aswinsuryan WDYT?

Reply:
this looks good , since it is listing all the configmaps in DefaultCoreDNSNamespace , we should be safe as there should be only one coredns final normally in this namespace.